test/tests/parallel/set2/test_helper.bats

```diff
@@ -130,43 +140,55 @@ load 'test_helper/common'
   assert_failure
 }
 
-@test "container_has_service_running/wait_for_service" {
+@test "${TEST_NAME_PREFIX} container_has_service_running/wait_for_service" {
+  # the container name is used here explicitly, because it cannot be a local
+  # variable as the `teardown` function would not have the correct scope
+  # and does not see the variable
+
   local PRIVATE_CONFIG
-  PRIVATE_CONFIG=$(duplicate_config_for_container .)
+  PRIVATE_CONFIG=$(duplicate_config_for_container . dms-test-services_running)
 
   # variable not local to make visible to teardown
-  CONTAINER_NAME=$(docker run -d --rm \
+  docker run -d --rm --name dms-test-services_running \
     -v "${PRIVATE_CONFIG}":/tmp/docker-mailserver \
     -h mail.my-domain.com \
-    -t "${NAME}")
+    -e ENABLE_CLAMAV=0 \
+    -e ENABLE_AMAVIS=0 \
+    -e ENABLE_FAIL2BAN=0 \
+    -e ENABLE_SPAMASSASIN=0 \
+    -t "${IMAGE_NAME}"
 
-  teardown() { docker rm -f "${CONTAINER_NAME}"; }
+  teardown() { docker rm -f dms-test-services_running; }
+
+  docker exec dms-test-services_running setup email add 'example_user@domain.com' 'password'
+  wait_for_finished_setup_in_container dms-test-services_running
+  wait_for_changes_to_be_detected_in_container dms-test-services_running
```

Comment:
Likewise, is this related to using a different `PRIVATE_CONFIG` for running in parallel?

Your additions here are a bit more extensive than the previous test (_although your addition there had `wait_for_finished_setup_in_container` both before and after it which seemed a bit odd_)

Reply:
Yes, this is related to the new `PRIVATE_CONFIG`. The additions here are those used to check whether the container has started properly - the other test for which I added these lines were checking the startup itself, hence less changes from my side with these tests.

Reply:
Reverted this change.